Lecture1.hs

```diff
@@ -0,0 +1,96 @@
+{-# OPTIONS_GHC -Wno-incomplete-patterns #-}
+module Lecture1
+(--makeSnippet
+ sumOfSquares
+, lastDigit
+, minmax
+, subString
+, strSum
+, lowerAndGreater
+, increase
+) where
+
+
+
+
+increase :: Integer -> Integer
+increase x = x + 1
+
+sumOfSquares :: Integer -> Integer -> Integer
+sumOfSquares n1 n2 = ( n1 ^ 2 ) + (n2 ^ 2)
+
+--finding last digit = finding the remainder when this number is divided by 10
+lastDigit :: Integer -> Integer
+lastDigit n = l
+ where
+     l = mod n 10
+
+minmax :: Integer -> Integer -> Integer -> Integer
+minmax n1 n2 n3 = l1 - l2
+    where
+       l = [n1, n2, n3]
+       l1 = maximum l
+       l2 = minimum l
+
+subString :: Int -> Int -> [Char] -> [Char]
+subString n1 n2 s1 = s
+    where
+        s2 = drop n1 s1
+        s = take (n2-n1) s2
+
+strSum :: String -> Int
+strSum s = n
+    where
+        s1 = words s
+        intList = map (read::String -> Int) s1
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
        intList = map read s1
```